src/Modules/CalcSetup.lua

```diff
@@ -991,7 +991,9 @@ function calcs.initEnv(build, mode, override, specEnv)
 					end
 					env.itemModDB:ScaleAddList(srcList, scale)
 				elseif item.type == "Quiver" and items["Weapon 1"] and items["Weapon 1"].name:match("Widowhail") then
-					scale = scale * (1 + (items["Weapon 1"].baseModList:Sum("INC", nil, "EffectOfBonusesFromQuiver") or 100) / 100)
+					local widowHailMod=(1 + (items["Weapon 1"].baseModList:Sum("INC", nil, "EffectOfBonusesFromQuiver") or 100) / 100)
+					scale = scale * widowHailMod
+					env.widowHailModifier = widowHailMod
```

Comment:
Nitpicking but not sure how i feel about adding modifiers directly to the env table. Something like this would be a bit cleaner imo:

```diff
diff --git a/src/Modules/CalcPerform.lua b/src/Modules/CalcPerform.lua
index 1bc2ad95c..9a591e90c 100644
--- a/src/Modules/CalcPerform.lua
+++ b/src/Modules/CalcPerform.lua
@@ -1005,7 +1005,7 @@ function calcs.perform(env, skipEHP)
                                env.minion.modDB:AddList(env.player.itemList["Weapon 1"].slotModList[1])
                        end
                        if env.player.itemList["Weapon 2"] and env.player.itemList["Weapon 2"].type == "Quiver" then
-                               env.minion.modDB:AddList(env.player.itemList["Weapon 2"].modList)
+                               env.minion.modDB:ScaleAddList(env.player.itemList["Weapon 2"].modList, m_max(modDB:Sum("BASE", nil, "WidowHailMultiplier"), 1))
                        end
                end
                if env.minion.itemSet or env.minion.uses then
diff --git a/src/Modules/CalcSetup.lua b/src/Modules/CalcSetup.lua
index 560b8c9d6..602475497 100644
--- a/src/Modules/CalcSetup.lua
+++ b/src/Modules/CalcSetup.lua
@@ -991,7 +991,9 @@ function calcs.initEnv(build, mode, override, specEnv)
                                        end
                                        env.itemModDB:ScaleAddList(srcList, scale)
                                elseif item.type == "Quiver" and items["Weapon 1"] and items["Weapon 1"].name:match("Widowhail") then
-                                       scale = scale * (1 + (items["Weapon 1"].baseModList:Sum("INC", nil, "EffectOfBonusesFromQuiver") or 100) / 100)
+                                       local widowHailMod=(1 + (items["Weapon 1"].baseModList:Sum("INC", nil, "EffectOfBonusesFromQuiver") or 100) / 100)
+                                       scale = scale * widowHailMod
+                                       env.modDB:NewMod("WidowHailMultiplier", "BASE", widowHailMod, "Widowhail")
                                        local combinedList = new("ModList")
                                        for _, mod in ipairs(srcList) do
                                                combinedList:MergeMod(mod)

```

Reply:
I'm fine with either, maintainers can decide. I am not deep enough into POB dev to know which one is more clean :)